ClientTests/HistoryDeletionUtilityTests.swift

```diff
@@ -0,0 +1,260 @@
+        XCTAssertTrue(metadataItems.isSuccess)
+        XCTAssertNotNil(metadataItems.successValue)
+        XCTAssertEqual(metadataItems.successValue!.count, sites.count)
+
+        let historyItems = profile.history.getSitesByLastVisit(limit: 100, offset: 0).value
+        XCTAssertTrue(historyItems.isSuccess)
+        XCTAssertNotNil(historyItems.successValue)
+        XCTAssertEqual(historyItems.successValue!.count, sites.count)
+
+        for (index, site) in sites.enumerated() {
+            guard let metadataURL = metadataItems.successValue?[index].url,
+                  let metadataTitle = metadataItems.successValue?[index].title?.lowercased(),
+                  let historyURL = historyItems.successValue?[index]?.url,
+                  let historyTitle = historyItems.successValue?[index]?.title.lowercased()
+            else {
+                XCTFail("Items that should exist in the database, do not.")
+                return
+            }
+
+            XCTAssertEqual(metadataURL, "https://www.\(site.domain).com/\(site.path)")
+            XCTAssertEqual(metadataTitle, "\(site.domain) test")
+            XCTAssertEqual(metadataItems.successValue![index].documentType, DocumentType.regular)
+            XCTAssertEqual(metadataItems.successValue![index].totalViewTime, 1)
+
+            XCTAssertEqual(historyURL, "https://www.\(site.domain).com/\(site.path)")
+            XCTAssertEqual(historyTitle, "\(site.domain) test")
+        }
+    }
+
+    private func populateDBHistory(with entries: [SiteElements]) {
+        entries.forEach { entry in
+            let site = createWebsiteFor(domain: entry.domain, with: entry.path)
+            addToLocalHistory(site: site)
+            setupMetadataItem(forTestURL: site.url,
+                              withTitle: site.title,
+                              andViewTime: 1)
+        }
+    }
+
+    private func createWebsiteFor(domain name: String, with path: String = "") -> Site {
+        let urlString = "https://www.\(name).com/\(path)"
+        let urlTitle = "\(name) test"
+
+        return Site(url: urlString, title: urlTitle)
+    }
+
+    private func addToLocalHistory(site: Site) {
+        let visit = SiteVisit(site: site, date: Date.nowMicroseconds())
+        XCTAssertTrue(profile.history.addLocalVisit(visit).value.isSuccess, "Site added: \(site.url).")
+    }
+
+    private func setupMetadataItem(forTestURL siteURL: String, withTitle title: String, andViewTime viewTime: Int32) {
+        let metadataKey1 = HistoryMetadataKey(url: siteURL, searchTerm: title, referrerUrl: nil)
+
+        XCTAssertTrue(profile.places.noteHistoryMetadataObservation(
+            key: metadataKey1,
+            observation: HistoryMetadataObservation(
+                url: metadataKey1.url,
+                viewTime: nil,
+                documentType: nil,
+                title: title
+            )
+        ).value.isSuccess)
```

Comment:
Could pass in `file: StaticString = #file, line: UInt = #line` in the function and set it to the assertions in there, so if there's a failure we're directed to the failing test automatically and don't have to search for it (same comment for other helper functions assertions).